src/main/kotlin/org/b12x/gfe/core/controller/version/LocalVersions.kt

```diff
@@ -0,0 +1,95 @@
+package org.b12x.gfe.core.controller.version
+
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import org.b12x.gfe.core.model.DataDownload
+import org.b12x.gfe.core.model.parsers.ParserVersionData
+import org.b12x.gfe.utilities.DirectoryManagement
+import org.b12x.gfe.utilities.FileManagement
+import java.io.File
+import java.nio.file.Paths
+import kotlin.io.path.exists
+
+class LocalVersions(loci: String) {
+
+    private val directoryManagement = DirectoryManagement()
+    private val gsgDataLocation = directoryManagement.setLociLocation(loci)
+    private val headerLength = 50  // bytes
+
+    val versionsList = createVersions()
+
+    /**
+     * Gets all versions for a specific set of genes in the user's GSG data directory.
+     *
+     * @return a list of versions available locally
+     */
+    fun createVersions(): ArrayList<Version> {//ArrayList<String> {
+
+        val versions = ArrayList<Version>()
+        // if the folder doesn't exist, there's nothing to add.
+        if (Paths.get(gsgDataLocation).exists() and File(gsgDataLocation).isDirectory) {
+            File(gsgDataLocation).listFiles().forEach {
+
+                // make sure we're looking at folders, that they have files in them
+                // and what they have in them actually contains data
+                if (it.exists() and it.isDirectory and folderContainsData(it)) {
+                    val locusAvailable = getLocusAvailable(it)
+                    val version = Version(it, it.name.toString(), locusAvailable)
+                    versions.add(version)
+                }
+            }
+        }
+        return versions
+    }
+
+    /**
+     * Gets all available locus names for a specific version
+     * Gets all files for a specific version of genes in the user's GSG data directory.
```

Comment:
Looks like there might have been a copy-paste error.

Reply:
It's not a copy paste error, it was deliberate, but it's a mistake because the second sentence is talking about HOW it's achieving this task, and we're not supposed to list that in Dokka. Second line deleted.